### DISCUSSION TITLE
Bankruptcy / Mark All Read button

### DISCUSSION THREAD

**Discussion by gigawhitlocks** [mattermost/community#22] [General Ideas]
Sometimes folks come back to a server after a long time and have way more unread messages than they're going to read. A Team-wide or Server-wide "bankruptcy" button that allows the user to "declare bankruptcy" and mark all messages as read could be useful.

**Comment by esethna** [comment]
Lol, yes! We have a ticket for this actually, it's much needed! The thought was to use a keyboard shortcut (Shift+Esc) with confirmation modal following similar design pattern in slack.

https://mattermost.atlassian.net/browse/MM-2541

**Reply by gigawhitlocks** [reply]
If I could come up with an idea that doesn't have a pre-existing ticket that'd be nice :sweat_smile:

**Comment by larkox** [comment]
I heard this idea before, and I remember an extra flavour to it. And that is that you can mark as read all messages until certain point. e.g. I am on PTO for a month. I don't want to read all the messages from the last month. But I want to know what has been happening lately. So I mark everything until two days ago as read. I still have stuff unread, but it is stuff that has been posted in the last two days, not in the last month I've been out.
